pdns/recursordist/rec-taskqueue.cc

```diff
@@ -70,29 +71,29 @@ static void resolve(const struct timeval& now, bool logErrors, const pdns::Resol
 
 void runTaskOnce(bool logErrors)
 {
-  t_taskQueue.runOnce(logErrors);
+  s_taskQueue.lock()->runOnce(logErrors);
```

Comment:
Reviewing my own code, I think this should be done differently. The lock should not be help while running the task.
It should be seomthign like this:
```
auto task = s_taskQueue.lock()->pop();
if (!task.empty()) {
  task.run();
}
```
plus some code if ther is no task available.